widgets/common/src/DynamicDropdown.jsx

```diff
@@ -147,24 +232,33 @@ export default function DynamicDropdown({
                 value={getDropdownValue()}
                 id={id}
                 name={name}
-                onChange={(event, field) => onChange(!_.isEmpty(field.value) ? field.value : null)}
-                onSearchChange={(e, data) => setSearchQuery(data.searchQuery)}
+                allowAdditions={allowAdditions}
+                onAddItem={(event, data) => {
+                    if (allowAdditions) setOptions([{ [valueProp]: data.value }, ...options]);
```

Comment:
Is there a point in checking if additions are allowed here? Does Semantic UI still fire `onAddItem` even if `allowAdditions={false}`?

Optional:

```suggestion
                    setOptions([{ [valueProp]: data.value }, ...options]);
```

Reply:
Make sense, will remove it.